### ISSUE TITLE
Session expiry

### ISSUE THREAD

**Issue by pranabsarma18** [csev/cc4e#92]
The login session expiry duration for www.cc4e.com is very short. While working on my assignment, I encountered an issue where I lost all my code because of the session expiry. I believe there should be a mechanism in place to automatically save the code every few seconds to prevent such incidents. Thank you.

**Comment by csev** [comment]
So I checked things out and the session duration is not the problem.  As long as you don't close your laptop, your session will be extended indefinitely.  However, there is not an auto-save and no Save button.  Let me think a bit about the best way to go.   Do you think an autosave every minute or so is the best idea?

**Comment by pranabsarma18** [comment]
> So I checked things out and the session duration is not the problem.  As long as you don't close your laptop, your session will be extended indefinitely.  However, there is not an auto-save and no Save button.  Let me think a bit about the best way to go.   Do you think an autosave every minute or so is the best idea?

Yes, I think autosave will be a good idea because sometimes while working on an assignment, I might want to take a break and come back again from where I left off with a fresh mind.
